lua/neotest-vstest/init.lua

```diff
@@ -78,10 +90,15 @@ function DotnetNeotestAdapter.root(path)
 end
 
 function DotnetNeotestAdapter.is_test_file(file_path)
+  if is_test_dir_cache[file_path] ~= nil then
+    return is_test_dir_cache[file_path]
+  end
```

Comment:
Have you observed that `is_test_file` are being called multiple times on your end? On my machine (and my assumption of how neotest is supposed to work) I only see this function being called the first time test discovery is triggered for a specific project.

Reply:
I think so, but it could also be that there was some other things that made things not work as expected for me. I have gone a few iterations on this, and it might not have any effect to have this. It's mostly the `filter_dir` which is necessary to avoid compilation of unwanted projects.

I can have a second look on this, and probably remove this

Reply:
Let’s test it out when the rest of the improvements are done and if we observe it’s being called multiple times then let’s keep the cache 😄

Reply:
I agree! Removed in latest commit